### DISCUSSION TITLE
Improve `astro:build:done` hooks, remove `RouteData`

### DISCUSSION THREAD

**Discussion by Jelenkee** [withastro/roadmap#647] [Proposal]
### What version of `astro` are you using?

1.9.1

### Are you using an SSR adapter? If so, which one?

no

### What package manager are you using?

npm

### What operating system are you using?

Linux

### Describe the Bug

When printing out the routeData for a dynamic route (e.g. _test-[id].astro_), a url for a specific page is printed.
![image](https://user-images.githubusercontent.com/59470612/211211324-bb1e4201-e04a-4a87-9c14-6aa9fb6886e8.png)

Perhaps you should remove the distURL for dynamic routes. Or even better, add all URLs for the route to the routeData

### Link to Minimal Reproducible Example

https://stackblitz.com/edit/github-9zdqva?file=astro.config.mjs,src%2Fpages%2Ftest-[id].astro&on=stackblitz

### Participation

- [X] I am willing to submit a pull request for this issue.

**Comment by florian-lefebvre** [comment]
`routes` on this hook have been deprecated, you should now use `assets`: https://docs.astro.build/en/reference/integrations-reference/#assets-option
